File: C++_Version/String/344Reverse_String.cpp
//Write a function that takes a string as input and returns the string reversed.
//
//Example:
//Given s = "hello", return "olleh".

#include <vector>
#include <iostream>
#include <algorithm>
#include <cstdlib>
#include <functional>
#include <string>
#include <unordered_map>
#include <map>
#include <set>
#include <unordered_set>
#include <cctype>
using namespace std;

class Solution {
public:
	string reverseString(string s) {
		reverse(s.begin(), s.end());

		return s;
	}
};

int main()
{
	Solution sol1;

	string A("hello"), B("baaabbbababaaabbbababaaabbbababaaabbbababaaabbbababaaabbbababaaabbbababaaabbbababaaabbbababaaabbbaba");

	string receive = sol1.reverseString(A);

	system("pause");

	return 0;
}

File: Java_Version/344Reverse_String.java
/*Write a function that takes a string as input and returns the string reversed.

        Example:
        Given s = "hello", return "olleh".*/


import java.lang.System;


public class Main
{
    public static void main(String[] args)
    {
        String input1 = "aba";
        String input2 = "hello";

        Solution solution = new Solution();

        String receiveFlag = solution.reverseString(input2);

        System.out.println(receiveFlag);
    }


}


class Solution {
    public String reverseString(String s) {

        StringBuilder stringBuilder = new StringBuilder(s);

        return stringBuilder.reverse().toString();
    }
}